source/fundamentals/indexes.txt

```diff
@@ -0,0 +1,253 @@
+   :language: java
+   :dedent:
+   :start-after: begin covered multikey query
+   :end-before: end covered multikey query
+
+Multikey indexes behave differently from non-multikey indexes in terms of query coverage, index bound computation, and
+sort behavior. For a full explanation of multikey indexes, including a discussion of their behavior and limitations,
+refer to the :manual:`Multikey Indexes page </core/index-multikey>` in the MongoDB manual.
+
+Text Indexes
+~~~~~~~~~~~~
+
+**Text indexes** support text search queries on string content. These indexes can include any field whose value is a
+string or an array of string elements. MongoDB supports text search for various languages. You can specify the default
+language as an option when creating the index.
+
+The following example uses the ``createIndex()`` method to create a ``text`` index on the ``fullplot`` field in the
+``movies`` collection in the ``sample_mflix`` database.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin text index
+   :end-before: end text index
+
+The following is an example of a query that would be covered by the index created above. Note that the ``sort`` is
+omitted because text indexes do not contain sort order.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin text query
+   :end-before: end text query
+
+For a full explanation of text search with MongoDB, refer to :manual:`Text Indexes </core/index-text>` in the MongoDB
+manual.
+
+Geospatial Indexes
+~~~~~~~~~~~~~~~~~~
+
+MongoDB supports queries of geospatial coordinate data using **2dsphere indexes**. With a 2dsphere index, you can query
+the geospatial data for inclusion, intersection, and proximity. For more information on querying geospatial data, see
+:manual:`Geospatial Queries </geospatial-queries/>`.
+
+To create a 2dsphere index, you must specify a field that contains only **GeoJSON objects**. For more details on this
+type, see the MongoDB server manual page on :manual:`GeoJSON objects </reference/geojson>`.
+
+The ``location.geo`` field in following sample document from the ``theaters`` collection in the ``sample_mflix``
+database is a GeoJSON Point object that describes the coordinates of the theater:
+
+.. code-block:: javascript
+
+   {
+      "_id" : ObjectId("59a47286cfa9a3a73e51e75c"),
+      "theaterId" : 104,
+      "location" : {
+         "address" : {
+            "street1" : "5000 W 147th St",
+            "city" : "Hawthorne",
+            "state" : "CA",
+            "zipcode" : "90250"
+         },
+         "geo" : {
+            "type" : "Point",
+            "coordinates" : [
+               -118.36559,
+               33.897167
+            ]
+         }
+      }
+   }
+
+The following example uses the ``createIndexes()`` method to create a ``2dsphere`` index on the ``location.geo`` field
```

Comment:
I noticed that index names are sometimes rendered in monospace text, sometimes plaintext. I think it makes sense for the "text" index above since that's the literal method name you use to create the index, but that's not the case here. Should we use "geo2dsphere" in monospace here instead? Ditto for some of the other index sections regarding monospacing vs. plaintext.